server/rpc.go

```diff
@@ -50,6 +51,85 @@ type Broadcaster interface {
 	Sign([]byte) ([]byte, error)
 }
 
+type BroadcastSessionsManager struct {
+	broadcastSessions []*BroadcastSession
+	broadcastSessMap  map[string]*BroadcastSession
+	sessLock          *sync.Mutex
+}
+
+func (bsm *BroadcastSessionsManager) selectSession() *BroadcastSession {
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+
+	// minOrchs is maximum number of inflight sessions
+	minOrchs := int(HTTPTimeout / SegLen)
+	if len(bsm.broadcastSessions) < minOrchs {
+		bsm.checkStatus()
+	}
+
+	if len(bsm.broadcastSessions) <= 0 {
+		return nil
+	}
+
+	last := len(bsm.broadcastSessions) - 1
+	sess, sessions := bsm.broadcastSessions[last], bsm.broadcastSessions[:last]
+	bsm.broadcastSessions = sessions
+	return sess
+}
+
+func (bsm *BroadcastSessionsManager) removeSession(session *BroadcastSession) {
+	bsm.sessLock.Lock()
+	delete(bsm.broadcastSessMap, session.OrchestratorInfo.Transcoder)
+	bsm.sessLock.Unlock()
+}
+
+func (bsm *BroadcastSessionsManager) completeSessions(sess *BroadcastSession) {
```

Comment:
Perhaps `completeSession` would be a more accurate name

Reply:
I don't have a preference between these two. `completeSession` sounds good to me!